src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISDeployerBase.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting.IIS
+{
+    public class IISDeployerBase : ApplicationDeployer
+    {
+        public IISDeploymentParameters IISDeploymentParameters { get; }
+
+        public IISDeployerBase(DeploymentParameters deploymentParameters, ILoggerFactory loggerFactory)
+            : base(deploymentParameters, loggerFactory)
+        {
+            if (deploymentParameters is IISDeploymentParameters)
```

Comment:
What if it's not?

Reply:
We can't call RunWebConfigActions or RunServerConfigActions. Right now I check for null in those methods.

Reply:
ugh

Reply:
Is this what we are going with long-term?

Reply:
It's ugly for sure. I think it's probably worth it now to add the copy constructor.

Reply:
In this repo yes but in other repos no.